src/server/Privilege.ts

```diff
@@ -1,19 +1,35 @@
-import { Cosmetics } from "../core/CosmeticSchemas";
+import { Cosmetics, Pattern } from "../core/CosmeticSchemas";
 import { PatternDecoder } from "../core/PatternDecoder";
 
-export class PrivilegeChecker {
+export interface PrivilegeChecker {
+  isPatternAllowed(
+    base64: string,
+    flares: readonly string[] | undefined,
+  ): true | "restricted" | "unlisted" | "invalid";
+  isCustomFlagAllowed(
+    flag: string,
+    flares: readonly string[] | undefined,
+  ): true | "restricted" | "invalid";
+}
+
+export class PrivilegeCheckerImpl implements PrivilegeChecker {
+  private b64ToPattern: Record<string, Pattern> = {};
+
   constructor(
     private cosmetics: Cosmetics,
     private b64urlDecode: (base64: string) => Uint8Array,
-  ) {}
+  ) {
+    for (const pattern of Object.values(this.cosmetics.patterns)) {
```

Comment:
You can substantially reduce the garbage generated by not using `Object` methods to change the shape of data before iterating over it.
```suggestion
    for (const name in this.cosmetics.patterns) {
      const pattern = this.cosmetics.patterns[name];
```